ci/copy_doxygen_files.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+# Prepare the source tree for Doxygen, see .github/workflows/doxygen.yml
+
+set -ex
+
+# Load the list of modified files.
+git fetch
+git branch -lr
+FILES=$(git diff --name-only origin/master... src/include)
+
+# Create a new temp directory.
+mkdir src/include2
+
+# Copy modified files into it.
+for FILE in $FILES
+do
+    cp "$FILE" ./src/include2
+done
+
+# Move real path aside.
+mv src/include src/include.old
+
+# And place the modified ones back in place.
+mv src/include2 src/include
+
+# Ensure there is at least one file present.
+if [ -f src/include.old/daos.h ]
+then
+    mv src/include.old/daos.h src/include
+fi
```

Comment:
You can use the array length to test this:

```suggestion
if [ ${#FILES[@]} -gt 0 ]; then
    mv src/include.old/daos.h src/include
fi
```

But rather than just making sure at least one file is present, if none are present (so none were modified) why not just exit here and add some kind of `if:` check to determine if `doxygen` should even run.

As an aside, https://github.com/tj-actions/changed-files looks like it might be useful in this workflow also.